From eb453bd1c37f9f023509ed28dda3a7cd1185de84 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 14:34:57 -0400
Subject: [scsi] Make sure cmd_flags are 64-bit

Message-id: <1402670238-13916-10-git-send-email-snitzer@redhat.com>
Patchwork-id: 83767
O-Subject: [RHEL7.1 PATCH 009/150] scsi: Make sure cmd_flags are 64-bit
Bugzilla: 1109271
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>
RH-Acked-by: Ewan Milne <emilne@redhat.com>

BZ: 1109271

Upstream commit 2bfad21ecc6f837de29743f4419f47dee3fac9e2
Author: Martin K. Petersen <martin.petersen@oracle.com>
Date:   Wed Apr 9 22:20:48 2014 -0400

    scsi: Make sure cmd_flags are 64-bit

    cmd_flags in struct request is now 64 bits wide but the scsi_execute
    functions truncated arguments passed to int leading to errors. Make sure
    the flags parameters are u64.

    Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>
    Cc: Jens Axboe <axboe@fb.com>
    CC: Jan Kara <jack@suse.cz>
    Cc: Frederic Weisbecker <fweisbec@gmail.com>
    Signed-off-by: Jens Axboe <axboe@fb.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/scsi_lib.c b/drivers/scsi/scsi_lib.c
index c49e91d..65d2b01 100644
--- a/drivers/scsi/scsi_lib.c
+++ b/drivers/scsi/scsi_lib.c
@@ -229,7 +229,7 @@ void scsi_queue_insert(struct scsi_cmnd *cmd, int reason)
  */
 int scsi_execute(struct scsi_device *sdev, const unsigned char *cmd,
    int data_direction, void *buffer, unsigned bufflen,
-   unsigned char *sense, int timeout, int retries, int flags,
+   unsigned char *sense, int timeout, int retries, u64 flags,
    int *resid)
 {
  struct request *req;
@@ -280,7 +280,7 @@ EXPORT_SYMBOL(scsi_execute);
 int scsi_execute_req_flags(struct scsi_device *sdev, const unsigned char *cmd,
        int data_direction, void *buffer, unsigned bufflen,
        struct scsi_sense_hdr *sshdr, int timeout, int retries,
-       int *resid, int flags)
+       int *resid, u64 flags)
 {
  char *sense = NULL;
  int result;
diff --git a/include/scsi/scsi_device.h b/include/scsi/scsi_device.h
index 0cf7b6d..6a1a4b7 100644
--- a/include/scsi/scsi_device.h
+++ b/include/scsi/scsi_device.h
@@ -456,11 +456,11 @@ extern int scsi_is_target_device(const struct device *);
 extern int scsi_execute(struct scsi_device *sdev, const unsigned char *cmd,
    int data_direction, void *buffer, unsigned bufflen,
    unsigned char *sense, int timeout, int retries,
-   int flag, int *resid);
+   u64 flags, int *resid);
 extern int scsi_execute_req_flags(struct scsi_device *sdev,
  const unsigned char *cmd, int data_direction, void *buffer,
  unsigned bufflen, struct scsi_sense_hdr *sshdr, int timeout,
- int retries, int *resid, int flags);
+ int retries, int *resid, u64 flags);
 static inline int scsi_execute_req(struct scsi_device *sdev,
  const unsigned char *cmd, int data_direction, void *buffer,
  unsigned bufflen, struct scsi_sense_hdr *sshdr, int timeout,
-- 
1.7.1